### ISSUE TITLE
Spacing between border and color when define textStyle

### ISSUE THREAD

**Issue by DFelten** [daohoangson/flutter_widget_from_html#515]
There is currently a problem when define a border and background together with a text style for the `HtmlWidget`. Spacing is inserted above and below the box between the background and the border. When removing the custom text style from the widget, it's working. 

In Flutter

![image](https://user-images.githubusercontent.com/1169185/117843171-e276ec00-b27e-11eb-8f1f-6f2932d3be26.png)

Our text widget

```dart
class TestHtmlWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const String text = '''
<div class="alert alert-info">
<h2>Test</h2>
</div>''';

    return HtmlWidget(
      text,
      webView: true,
      textStyle: const TextStyle(fontSize: 14),
      factoryBuilder: () => TMXWidgetFactory(context: context),
    );
  }
}
```
The widget factory

```dart
class TMXWidgetFactory extends WidgetFactory {
  TMXWidgetFactory({required this.context});

  final BuildContext context;

  @override
  void parse(BuildMetadata meta) {
    final String className = meta.element.className;

    if (className.contains('alert')) {
      meta['padding'] = '12px';
    }

    if (className.contains('alert-info')) {
      meta['border'] = '2px solid #000000';
      meta['background-color'] = '#5cb85c';
    }

    super.parse(meta);
  }
}
```

In the browser it looks like this:

![image](https://user-images.githubusercontent.com/1169185/117843101-d25f0c80-b27e-11eb-84ce-b279f4f156b8.png)

```html
<!DOCTYPE html>
<html>
<head>
<style>
.alert-info {
  background: #5cb85c;
  border: 2px solid #000000;
}
</style>
</head>
<body>

  <div class="alert alert-info">
    <h2>Test</h2>
    </div>

</body>
</html>
```

**Comment by DFelten** [comment]
Thanks :) Unfortunately there is still a bug I think. 

The empty space now has the correct background color. Still, there is too much padding here.

![image](https://user-images.githubusercontent.com/1169185/118266143-797bb800-b4ba-11eb-80ff-34432cc89289.png)

![image](https://user-images.githubusercontent.com/1169185/118266109-7254aa00-b4ba-11eb-9138-f73618a5d7af.png)

**Comment by daohoangson** [comment]
> there is too much padding here.

What is your HTML?

**Comment by DFelten** [comment]
It's the same as shown above:

```dart
class TestHtmlWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const String text = '''
<div class="alert alert-info">
<h2>Test</h2>
</div>''';

    return HtmlWidget(
      text,
      webView: true,
      textStyle: const TextStyle(fontSize: 14),
      factoryBuilder: () => TestWidgetFactory(),
    );
  }
}
```

```dart
class TestWidgetFactory extends WidgetFactory {
  @override
  void parse(BuildMetadata meta) {
    final String className = meta.element.className;

    if (className.contains('alert')) {
      meta['padding'] = '12px';
    }

    if (className.contains('alert-info')) {
      meta['border'] = '2px solid #000000';
      meta['background-color'] = '#5cb85c';
    }

    super.parse(meta);
  }
}
```

![image](https://user-images.githubusercontent.com/1169185/118370175-d359ac00-b5a6-11eb-8c3b-3982175f0420.png)

Without the border

![image](https://user-images.githubusercontent.com/1169185/118370194-eff5e400-b5a6-11eb-8bcf-ed596f84670e.png)

**Comment by DFelten** [comment]
Ok, it seems the reason here is the h2. There is some default padding.

It's working when creating a customStylesBuilder for h2.

```dart
customStylesBuilder: (element) {
  if (element.localName == 'h2') {
    return {'padding': '0', 'margin': '0'};
  }

  return null;
},
```

**Comment by daohoangson** [comment]
Yeah, the `H2` tag has some default margin. I tried with this HTML to replicate your code:

```html
<div style="background: red">above</div>

<div style="padding: 12px; background: #5cb85c;">
  <h2>Test</h2>
</div>

<div style="background: red">mid</div>

<div style="padding: 12px; border: 2px solid #000000; background: #5cb85c;">
  <h2>Test</h2>
</div>

<div style="background: red">below</div>
```

And compared between browser rendering and this package's, I got this:

| Browser | Package |
| --- | --- |
| ![](https://user-images.githubusercontent.com/239336/118370769-72949880-b5d3-11eb-93c7-ebd983562fb6.png) | ![](https://user-images.githubusercontent.com/239336/118370767-6f99a800-b5d3-11eb-87ff-012e4a911917.png) |

So there is a bug using padding+background together...

**Comment by DFelten** [comment]
Ok, then I think that would be a new ticket. The problem here with the text styles and the background is fixed.

**Comment by daohoangson** [comment]
Yeah, I'm working on it here #523.

**Comment by daohoangson** [comment]
It has been merged @DFelten. Let me know if a new release is required.

**Comment by DFelten** [comment]
Thanks for the great support! Our next release will take a while, so it's not urgent at the moment. So we can wait a little bit here to see if more changes will be added. After all, every release is more work for you :)

**Comment by daohoangson** [comment]
Since the fix has been released in `v0.7.0-dev.2021052101`, I'm closing this now.
